### PR TITLE
Add getImageSourceOptimistic

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Adds `getImageSourceOptimistic` that will return the first image source for a given user and cid. This can be used to get the image source for an array of tracks for example for the react-native-track-player

### Dragons

`useLocalImage` is still a hook so we can't support local images in the track-player yet, but can add that next

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/audio/Audio.tsx [hunk 1]
```diff
@@ -447,9 +443,21 @@ export const Audio = () => {
     }
 
     currentUriRef.current = newUri
-    const imageUrl = trackImageSource?.source?.[2]?.uri ?? DEFAULT_IMAGE_URL
+    const imageUrl =
+      getImageSourceOptimistic({
```
Comment (dylanjeffers) [inline packages/mobile/src/components/audio/Audio.tsx:447]: this is great!

File: packages/mobile/src/components/audio/Audio.tsx [hunk 2]
```diff
@@ -447,9 +443,21 @@ export const Audio = () => {
...
+        cid: track ? track.cover_art_sizes || track.cover_art : null,
+        user: trackOwner
+        // localSource
```
Comment (dylanjeffers) [inline packages/mobile/src/components/audio/Audio.tsx:450]: comment

File: packages/mobile/src/hooks/useContentNodeImage.ts [hunk 3]
```diff
@@ -58,6 +58,65 @@ const createImageSourcesForEndpoints = ({
...
+  ]
+  return sourceList
+}
+
+/**
+ * Return the first image source, usually the user's primary
+ * or a local source. This is useful for cases where there is no error
+ * callback if the image fails to load - like the MusicControls on the lockscreen
+ */
+export const getImageSourceOptimistic = (
```
Comment (dylanjeffers) [inline packages/mobile/src/hooks/useContentNodeImage.ts:113]: love the reuse, nice

**Review by dylanjeffers — APPROVED** [review]
nice job

File: packages/mobile/src/components/audio/Audio.tsx (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/components/audio/Audio.tsx:447]: Gonna leave this in as an indicator that we still need to make this work with localSource

**Review by Kyle-Shanks — APPROVED** [review]
Nice!

**Pull request merged**